### DISCUSSION TITLE
Linking Static Target SU2_CFD

### DISCUSSION THREAD

**Discussion by okaaany** [su2code/SU2#2302] [Q&A] [answered]
I have been trying to activate Python Wrapper in SU2 for a while. I have installed packages such as open-mpi, swig, mpi4py, gcc, g++.

I just want to use Python scripts with SU2 and i have been trying to solve this issue for over a month now.

I ran _"./meson.py build -Denable-pywrapper=true --prefix=/home/yaman/Masaüstü/SU2"._

I made the necessary adjustments to my .bashrc file:
_"        export SU2_RUN=/home/yaman/Masaüstü/SU2/bin
        export SU2_HOME=/home/yaman/Masaüstü/SU2
        export PATH=$PATH:$SU2_RUN
        export PYTHONPATH=$PYTHONPATH:$SU2_RUN        "_

Finally, after the code _"./meson.py build --reconfigure -Dstatic-cgns-deps=true",_

I moved on to the _"./ninja -C build install"_ step.

I should mention the error:

_"[351/814] Compiling C object externals/cgns/libcgns.a.p/adf_ADF_internals.c.o
FAILED: externals/cgns/libcgns.a.p/adf_ADF_internals.c.o
cc -Iexternals/cgns/libcgns.a.p -Iexternals/cgns -I../externals/cgns -Iexternals/cgns/hdf5 -I../externals/cgns/hdf5 -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -std=c99 -O3 -fPIC -Wno-unused-result -Wno-unused-parameter -Wno-unused-variable -Wno-unused-but-set-variable -Wno-sign-compare -Wno-error=unused-function -Wno-pedantic -Wno-error=stringop-truncation -Wno-stringop-truncation -Wno-error=absolute-value -Wno-error=class-memaccess -MD -MQ externals/cgns/libcgns.a.p/adf_ADF_internals.c.o -MF externals/cgns/libcgns.a.p/adf_ADF_internals.c.o.d -o externals/cgns/libcgns.a.p/adf_ADF_internals.c.o -c ../externals/cgns/adf/ADF_internals.c
../externals/cgns/adf/ADF_internals.c: In function 'ADFI_open_file':
../externals/cgns/adf/ADF_internals.c:194:17: error: implicit declaration of function 'fileno' [-Wimplicit-function-declaration]
 194 | # define FILENO fileno
     |                 ^~~~~~
../externals/cgns/adf/ADF_internals.c:5531:32: note: in expansion of macro 'FILENO'
 5531 |    f_ret = ftmp == NULL ? -1 : FILENO(ftmp);
     |                                ^~~~~~
[364/814] Compiling C object externals/cgns/hdf5/libsu2hdf5.a.p/H5Tconv.c.o
ninja: build stopped: subcommand failed."_

Then I continued with _" ./ninja -C build install -k 0"._

**The last error I got was:**

_"[443/451] Linking static target SU2_CFD/src/libSU2core.a
ninja: build stopped: cannot make progress due to previous errors."_

    OS: [Arch Linux (2024.06.01)]
    C++ compiler and version: [ g++ (GCC) 14.1.1]
    MPI implementation and version: [Open-MPI 5.0.3]
    SU2 Version: [8.0.1]

**Comment by pcarruscag** [comment, marked as answer]
Have you tried without CGNS? -Denable-cgns=false?

**Reply by okaaany** [reply]
Incredible, spot-on answer. Thank you so much. The installation completed successfully.

The final version of **User Defined options** was as follows:

  prefix         : _/home/yaman/Masaüstü/SU2_
  enable-cgns    : _false_
  enable-pywrapper: _true_
  static-cgns-deps: _true_

**Reply by pcarruscag** [reply]
well it was more a debug step, if you need support for CGNS meshes we need to make it work with -Denable-cgns=true, but if it works for you, problem solved :+1:
